bin/faucet/src/server/pow.rs

```diff
@@ -133,10 +142,36 @@ impl TryFrom<&RawMintRequest> for PowParameters {
                 .pow_solution
                 .as_ref()
                 .ok_or(InvalidRequest::MissingPowParameters)?,
+            difficulty: *value
+                .pow_difficulty
+                .as_ref()
+                .ok_or(InvalidRequest::MissingPowParameters)?,
         })
     }
 }
 
+// POW
+// ================================================================================================
+
+#[derive(Clone)]
+pub struct PoW {
+    pub(crate) salt: String,
+    pub(crate) difficulty: Arc<AtomicU64>,
+    pub(crate) challenge_cache: ChallengeCache,
+}
+
+impl PoW {
+    /// Adjust the difficulty of the `PoW`.
+    ///
+    /// The difficulty is adjusted based on the number of active requests.
+    /// The difficulty is increased by 1 for every 50 active requests.
+    /// The difficulty is clamped between 1 and `MAX_DIFFICULTY`.
+    pub fn adjust_difficulty(&self, active_requests: usize) {
```

Comment:
Right now if we go back (for example) to a max queue of 200 requests the effective max difficulty is 4, right? If not, disregard this comment. 

But otherwise, and maybe not for now, should this function take into account the max queue length somehow? 
For example, we could map the queue length into the difficulty range by computing the filled ration of the queue and applying that to the `[1..MAX_DIFFICULTY]` range. This way the step interval (50 in this case) is not "coupled" with the queue length in any way and we could tweak difficulty or max queue length and expect it to work linearly (or we could decide to grow it exponentially, etc.).

Reply:
I just saw that the difficulty is adjusted based on total requests (even if the PoW does not validate), not just minting requests in progress. I was expecting it to be over the requests that validated correctly, but I think this might be fine and maybe dismisses my comment above. Not entirely sure if there is a practical difference. 

In any case, would it make sense to make the step interval be configurable? Maybe as part of #878